pyreal/visualize/feature_based_vis.py

```diff
@@ -395,6 +395,8 @@ def feature_scatter_plot(
     if legend_type == "continuous":
         norm = plt.Normalize(0, 1)
         sm = plt.cm.ScalarMappable(cmap=PALETTE_CMAP, norm=norm)
+        if categorical:
```

Comment:
Test plan image looks good! I think it may be a bit easier to read if we align the right end of the text with the tick (at the moment, the tick is centered above the rotated label, which especially with longer labels may be confusing) I think just adding `ha='right'` to the xticks command should work